vault/seal/gcpckms/gcpckms.go

```diff
@@ -288,36 +273,15 @@ func (s *GCPCKMSSeal) Decrypt(_ context.Context, in *physical.EncryptedBlobInfo)
 	return plaintext, nil
 }
 
-func (s *GCPCKMSSeal) getClient() (*cloudkms.Service, error) {
-	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, cleanhttp.DefaultPooledClient())
-
-	var client *http.Client
-	// If the credentials path was provided explicitly then use that
-	if s.credsPath != "" {
-		creds, err := ioutil.ReadFile(s.credsPath)
-		if err != nil {
-			return nil, err
-		}
-
-		conf, err := google.JWTConfigFromJSON(creds, cloudkms.CloudPlatformScope)
-		if err != nil {
-			return nil, err
-		}
-
-		client = conf.Client(ctx)
-	} else {
-		// Otherwise use application default credentials
-		var err error
-		client, err = google.DefaultClient(ctx, cloudkms.CloudPlatformScope)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	kmsClient, err := cloudkms.New(client)
+func (s *GCPCKMSSeal) getClient() (*cloudkms.KeyManagementClient, error) {
+	client, err := cloudkms.NewKeyManagementClient(context.Background(),
+		option.WithCredentialsFile(s.credsPath),
```

Comment:
If creds aren't given (i.e. ""), this falls back to default app credentials as was true with the previous behavior. If it's provided, the client handles reading the file from disk and extracting the credential.